src/wallet/bdb.cpp

```diff
@@ -623,8 +623,8 @@ bool BerkeleyDatabase::PeriodicFlush()
     if (!lockDb) return false;
 
     // Don't flush if any databases are in use
-    for (auto it = env->mapFileUseCount.begin() ; it != env->mapFileUseCount.end(); it++) {
-        if ((*it).second > 0) return false;
+    for (const auto& use_count : env->mapFileUseCount) {
+        if (use_count.second > 0) return false;
```

Comment:
fa73493930e35850e877725167dc9d42e47015c8

nit, if you feel like cleaning it a bit more maybe do this below:
```cpp
if (env->mapFileUseCount.erase(strFile) == 0) return false
```
And drop `.erase(it)`.

Reply:
Thanks, but I'll leave that for a follow-up